packages/admin/cms-admin/src/blocks/table/__tests__/insertRow.spec.tsx

```diff
@@ -0,0 +1,51 @@
+import { fireEvent, within } from "@testing-library/react";
+import { useState } from "react";
+
+import { type TableBlockData } from "../../../blocks.generated";
+import { render } from "../../../testing/testUtils";
+import { mockTableData } from "../__mocks__/TableBlockData.mocks";
+import { TableBlockGrid } from "../TableBlockGrid";
+
+const getRenderedTableBlock = (initialState: TableBlockData) => {
+    const Component = () => {
+        const [state, setState] = useState<TableBlockData>(initialState);
+        return <TableBlockGrid state={state} updateState={setState} />;
+    };
+    return render(<Component />);
+};
+
+describe("TableBlock: Insert a new row", () => {
+    it("should insert a new, empty row at the top of the table", async () => {
+        const rendered = getRenderedTableBlock(mockTableData);
+
+        const rowOptionsButtons = rendered.getAllByLabelText(/row options/i);
+        const firstRowOptionsButton = rowOptionsButtons[0];
+        fireEvent.click(firstRowOptionsButton);
+
+        const insertRowAboveButton = rendered.getByRole("menuitem", { name: /add row above/i });
+        fireEvent.click(insertRowAboveButton);
+
+        const rowgroup = rendered.getByRole("rowgroup");
+        const rows = within(rowgroup).getAllByRole("row");
+
+        const numberOfRowsAfterInsertingOne = mockTableData.rows.length + 1;
+        expect(rows).toHaveLength(numberOfRowsAfterInsertingOne);
+
+        rows.forEach((row, rowIndex) => {
+            const cells = within(row).getAllByRole("gridcell");
+            const cellsExcludingTheDragHandleAndActionsCells = cells.slice(1, cells.length - 1);
+
+            cellsExcludingTheDragHandleAndActionsCells.forEach((cell, cellIndex) => {
+                const isNewlyInsertedRow = rowIndex === 0;
+                if (isNewlyInsertedRow) {
+                    expect(cell.textContent).toBe("");
+                } else {
+                    const rowIndexBeforeInsertingNewRow = rowIndex - 1;
+                    const valueOfCellAboveTheCurrentBeforeInsertingNewRow =
```

Comment:
```suggestion
                    const valueOfCellBeforeInsertingNewRow =
```

Ich verstehe nicht das "Above Current" part, ist es nicht eigentlich "Below Current Row" da du `rowIndex - 1` machst?

Anyways ich glaub das ist nicht nötig in dem Namen ^^ `valueOfCellBeforeInsertingNewRow` mMn